### PR TITLE
Scan on location change

### PR DESCRIPTION
**Pull request by arcturus**
@wilsonpage f? & r?

### CODE REVIEW

File: ios/LocationChangeReceiver.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  
+  @objc override init() {
+    super.init()
+    scanner = MagnetScanner(callback: self.onMagnetScanner)
+  }
+  
+  private func onMagnetScanner(result: Dictionary<String, AnyObject>) {
+    let url = result["url"] as! String
+    // Uncomment following line for debug
+    //NotificationsHelper.showNotification("Url we should notify \(url)")
```
Comment (arcturus) [inline ios/LocationChangeReceiver.swift]: As we don't always notify (if we saw the url in the past 10 minutes), this line is helpful to see if we discovered something while we were in background.

File: ios/Magnet.xcodeproj/xcshareddata/xcschemes/Magnet.xcscheme [hunk 2]
```diff
@@ -17,7 +17,7 @@
                BlueprintIdentifier = "13B07F861A680F5B00A75B9A"
                BuildableName = "Magnet.app"
                BlueprintName = "Magnet"
-               ReferencedContainer = "container:magnet.xcodeproj">
+               ReferencedContainer = "container:Magnet.xcodeproj">
```
Comment (arcturus) [inline ios/Magnet.xcodeproj/xcshareddata/xcschemes/Magnet.xcscheme:20]: Just realised this changes were automatically added by xcode :S

File: ios/Magnet/Images.xcassets/AppIcon.appiconset/Contents.json [hunk 3]
```diff
@@ -1,6 +1,16 @@
 {
   "images" : [
     {
+      "idiom" : "iphone",
```
Comment (arcturus) [inline ios/Magnet/Images.xcassets/AppIcon.appiconset/Contents.json:4]: Same than this ones.

File: ios/Magnet/Info.plist [hunk 4]
```diff
@@ -35,8 +35,10 @@
 			</dict>
 		</dict>
 	</dict>
+	<key>NSLocationAlwaysUsageDescription</key>
+	<string>Project Magnet needs to check your location</string>
 	<key>NSLocationWhenInUseUsageDescription</key>
-	<string></string>
+	<string>Project Magnet needs to check your location</string>
```
Comment (arcturus) [inline ios/Magnet/Info.plist:41]: Now we need to ask for 'always usage description', I'm open to better wording you know I'm not the best there :P

File: ios/Magnet/Info.plist [hunk 5]
```diff
@@ -45,12 +47,15 @@
...
 		<string>bluetooth-central</string>
+		<string>fetch</string>
+		<string>location</string>
 	</array>
 	<key>UILaunchStoryboardName</key>
 	<string>LaunchScreen</string>
 	<key>UIRequiredDeviceCapabilities</key>
 	<array>
 		<string>armv7</string>
+		<string>location-services</string>
```
Comment (arcturus) [inline ios/Magnet/Info.plist:58]: Didn't for the device to have gps, but to have location-services, meaning that ipod and ipads will still be working, but location will be restricted.

File: ios/MagnetScannerReact.swift [hunk 6]
```diff
@@ -26,39 +23,6 @@ import Foundation
       .sendDeviceEventWithName("magnetscanner:itemfound", body: item)
     
     let url = item["url"] as! String
-    notifyUser(url)
```
Comment (arcturus) [inline ios/MagnetScannerReact.swift:29]: Moved all this code to notifications, this was in charge of throttling  the notifications, so is better to live there, also used now for the geo notifications in bg.

**Review by arcturus — COMMENTED** [review]
In order to test this with the simulator you can go to the use xplanner:
https://github.com/zackslash/XJourney/
(follow the instructions about permissions)

Also I can provide with a csv file with a route through central london.

File: ios/LocationChangeReceiver.swift [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+@objc(LocationChangeReceiver) class LocationChangeReceiver: NSObject, CLLocationManagerDelegate {
+  let locationManager:CLLocationManager = CLLocationManager()
+  var scanner: MagnetScanner! = nil
+  
+  @objc override init() {
+    super.init()
+    scanner = MagnetScanner(callback: self.onMagnetScanner)
+  }
+  
+  private func onMagnetScanner(result: Dictionary<String, AnyObject>) {
```
Comment (wilsonpage) [inline ios/LocationChangeReceiver.swift]: `onItemFound` and `item`

**Comment by wilsonpage** [comment]
- [x] We're still seeing several requests to magnet-service instead of just one
- [x] Can you file a follow-up to pull all location logic into magnet-scanner-ios

**Comment by wilsonpage** [comment]
- [x] Please remove all debug comments

**Pull request merged**